symengine/pow.cpp

```diff
@@ -94,7 +94,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero) && is_a_Number(*b)) {
```

Comment:
But, it might create a problem. Like, in this case:
```
a = 0**b + 5
b = -1
print (a)
```
if we return `0**x` as 0, `a` will evaluate to 5.
Even in sympy they store it as `0**x`, and the output is `0**b + 5`.